x-pack/plugins/security_solution/public/management/pages/response_actions/view/response_actions_list.tsx

```diff
@@ -0,0 +1,475 @@
+import type { EndpointActionListRequestQuery } from '../../../../../common/endpoint/schema/actions';
+import { AdministrationListPage } from '../../../components/administration_list_page';
+import { ManagementEmptyStateWrapper } from '../../../components/management_empty_state_wrapper';
+import { useGetEndpointActionList } from '../../../hooks';
+import { OUTPUT_MESSAGES, TABLE_COLUMN_NAMES, UX_MESSAGES } from '../translations';
+import { MANAGEMENT_PAGE_SIZE_OPTIONS } from '../../../common/constants';
+import { useTestIdGenerator } from '../../../hooks/use_test_id_generator';
+
+// Truncated usernames
+const StyledFacetButton = styled(EuiFacetButton)`
+  .euiText {
+    margin-top: 0.38rem;
+    overflow-y: visible !important;
+  }
+`;
+
+export const ResponseActionsList = memo<
+  EndpointActionListRequestQuery & { hideHeader?: boolean; hideHostNameColumn?: boolean }
+>(
+  ({
+    agentIds,
+    commands,
+    endDate,
+    startDate,
+    userIds,
+    hideHeader = false,
+    hideHostNameColumn = false,
+  }) => {
+    const getTestId = useTestIdGenerator('response-actions-list');
+
+    const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<{
+      [k: ActionDetails['id']]: React.ReactNode;
+    }>({});
+
+    const [queryParams, setQueryParams] = useState<EndpointActionListRequestQuery>({
+      page: 0,
+      pageSize: 10,
+      agentIds,
+      commands,
+      endDate,
+      startDate,
+      userIds,
+    });
+
+    // initial fetch of list data
+    const {
+      error,
+      data: actionList,
+      isFetching,
+      isFetched,
+      refetch: reFetchEndpointActionList,
+    } = useGetEndpointActionList(queryParams, {
+      enabled: true,
+      refetchOnMount: true,
+      keepPreviousData: false,
+    });
+
+    // total actions
+    const totalItemCount = useMemo(() => actionList?.total, [actionList]);
+
+    // expanded tray contents
+    const toggleDetails = useCallback(
+      (item: ActionDetails) => {
+        const itemIdToExpandedRowMapValues = { ...itemIdToExpandedRowMap };
+        if (itemIdToExpandedRowMapValues[item.id]) {
+          delete itemIdToExpandedRowMapValues[item.id];
+        } else {
+          const {
+            startedAt,
+            completedAt,
+            isCompleted,
+            wasSuccessful,
+            isExpired,
+            command,
+            logEntries,
+          } = item;
+
+          const responseData = logEntries.reduce<
+            Array<{
+              data: LogsEndpointActionResponse['EndpointActions']['data'];
+              error: LogsEndpointActionResponse['error'];
+            }>
+          >((acc, curr) => {
+            if (curr.type === 'response') {
+              acc.push({
+                data: curr.item.data.EndpointActions.data,
+                error: curr.item.data.error,
+              });
+            }
+            return acc;
+          }, []);
+
+          const descriptionListItems = [
```

Comment:
Having all these items in the same array and doing a `slice` later make it difficult to read. Could you create different arrays with the desired values for each ` listItems` ?